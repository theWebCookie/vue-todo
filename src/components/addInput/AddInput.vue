<template>
  <div :class="$style.inputWrapper">
    <input type="checkbox" disabled :class="$style.checkBox" />
    <input type="text" placeholder="Add your activity" :class="$style.searchBar" v-model="inputValue" @keyup.enter="addActivity" />
  </div>
</template>

<script>
export default {
  name: 'AddInput',
  data() {
    return {
      inputValue: '',
    };
  },
  methods: {
    addActivity() {
      console.log(this.inputValue);
      this.inputValue = '';
    },
  },
};
</script>

<style module>
.inputWrapper {
  display: flex;
  gap: 10px;
  background-color: #fff;
  padding: 10px 0;
  margin-bottom: 20px;
  border-radius: 5px;
}

.checkBox {
  margin-left: 10px;
}

.searchBar {
  width: 100%;
  padding: 10px 0;
  font-size: 16px;
  border: none;
}
</style>
