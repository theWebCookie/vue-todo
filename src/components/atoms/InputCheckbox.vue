<template>
  <input type="checkbox" :disabled="disabled" :class="$style.checkBox" @change="toggleTask(id)" :checked="checked" />
</template>

<script>
export default {
  name: 'InputCheckbox',
  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
    checked: {
      type: Boolean,
      default: false,
    },
    id: {
      type: Number,
    },
  },
  methods: {
    toggleTask(taskId) {
      this.$store.dispatch('toggleTask', taskId);
    },
    removeTask(taskId) {
      this.$store.dispatch('removeTask', taskId);
    },
  },
};
</script>

<style module lang="scss">
.checkBox {
  margin-left: $spacing-small;
}
</style>
