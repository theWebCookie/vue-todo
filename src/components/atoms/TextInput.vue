<template>
  <input type="text" :placeholder="placeholder" :class="$style.searchBar" v-model="inputValue" @keyup.enter="addTask" />
</template>

<script>
export default {
  name: 'TextInput',
  props: {
    placeholder: {
      type: String,
      default: 'Add your task',
    },
  },
  data() {
    return {
      inputValue: '',
    };
  },
  methods: {
    addTask() {
      if (this.inputValue.trim()) {
        const task = {
          id: Date.now(),
          title: this.inputValue,
          completed: false,
        };
        this.$store.dispatch('addTask', task);
        this.inputValue = '';
      }
    },
  },
};
</script>

<style module lang="scss">
.searchBar {
  width: 100%;
  padding: $spacing-small 0;
  font-size: 16px;
  border: none;
}
</style>
