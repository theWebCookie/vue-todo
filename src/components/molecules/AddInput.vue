<template>
  <div :class="[$style.inputWrapper, exceedsLimit ? $style.exceededLimit : '']">
    <InputCheckbox :disabled="true" />
    <TextInput :placeholder="'Add your task'" @input-exceeds-limit="handleInputExceedsLimit" />
  </div>
</template>

<script>
import InputCheckbox from '../atoms/InputCheckbox.vue';
import TextInput from '../atoms/TextInput.vue';

export default {
  name: 'AddInput',
  components: {
    InputCheckbox,
    TextInput,
  },
  data() {
    return {
      exceedsLimit: false,
    };
  },
  methods: {
    handleInputExceedsLimit(value) {
      this.exceedsLimit = value;
    },
  },
};
</script>

<style module lang="scss">
.inputWrapper {
  display: flex;
  gap: $spacing-small;
  background-color: $white-color;
  padding: $spacing-small;
  margin-bottom: $spacing-large;
  border-radius: $border-radius;
  box-sizing: border-box;
}

.exceededLimit {
  border: 1px solid red;
  background-color: #ffe6e6;
}
</style>
