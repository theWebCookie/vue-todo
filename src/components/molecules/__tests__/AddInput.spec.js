import { shallowMount } from '@vue/test-utils';
import { describe, it, expect } from '@jest/globals';
import AddInput from '../AddInput.vue';
import InputCheckbox from '../../atoms/InputCheckbox.vue';
import TextInput from '../../atoms/TextInput.vue';

describe('AddInput.vue', () => {
  it('renders correctly and matches the snapshot', () => {
    const wrapper = shallowMount(AddInput, {
      global: {
        mocks: {
          $style: {
            inputWrapper: 'inputWrapper',
          },
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders InputCheckbox and TextInput components with default props', () => {
    const wrapper = shallowMount(AddInput, {
      global: {
        mocks: {
          $style: {
            inputWrapper: 'inputWrapper',
          },
        },
      },
    });

    const inputCheckbox = wrapper.findComponent(InputCheckbox);
    const textInput = wrapper.findComponent(TextInput);

    expect(inputCheckbox.exists()).toBe(true);
    expect(inputCheckbox.props('disabled')).toBe(true);
    expect(inputCheckbox.props('checked')).toBe(false);

    expect(textInput.exists()).toBe(true);
    expect(textInput.props('placeholder')).toBe('Add your task');
  });
});
