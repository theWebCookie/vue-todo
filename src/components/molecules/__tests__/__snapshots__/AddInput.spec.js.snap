// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddInput.vue renders correctly and matches the snapshot 1`] = `
<div
  class="inputWrapper"
>
  <input-checkbox-stub
    checked="false"
    disabled="true"
  />
  <text-input-stub
    placeholder="Add your task"
  />
</div>
`;
